Source/JavaScriptCore/runtime/JSObjectInlines.h

```diff
@@ -868,13 +868,69 @@ bool JSObject::fastForEachPropertyWithSideEffectFreeFunctor(VM& vm, const Functo
 
     Structure* structure = this->structure();
 
-    if (!structure->canPerformFastPropertyEnumeration())
+    if (!structure->canPerformFastPropertyEnumerationCommon())
         return false;
 
     structure->forEachProperty(vm, functor);
     return true;
 }
 
+template<typename Functor>
+void JSObject::forEachIndexedProperty(JSGlobalObject* globalObject, const Functor& functor)
+{
```

Comment:
Add `ASSERT(structure()->canPerformFastPropertyEnumerationCommon());` This should be called only when it is true.